php_phongo.c

```diff
@@ -738,11 +742,13 @@ int phongo_execute_query(mongoc_client_t *client, const char *namespace, zval *z
 	if (query->read_concern) {
 		mongoc_collection_set_read_concern(collection, query->read_concern);
 	}
+	
+	opts = bson_new();
 
 	/* FIXME: Legacy way of specifying the readPreference option into this function */
 	if (options && Z_TYPE_P(options) == IS_OBJECT && instanceof_function(Z_OBJCE_P(options), php_phongo_readpreference_ce TSRMLS_CC)) {
 		zreadPreference = options;
-	} else if (!phongo_execute_parse_options(client, server_id, options, PHONGO_COMMAND_READ, NULL, &zreadPreference, NULL TSRMLS_CC)) {
+	} else if (!phongo_execute_parse_options(client, server_id, options, PHONGO_COMMAND_READ, opts, &zreadPreference, NULL TSRMLS_CC)) {
```

Comment:
Same issues with `opts` as the bulk write fixes (never freed, and never used after options are parsed).

Unlike bulk write, we do have a way to pass opts in with the find command with `query->opts`. I think it would make more sense to pass that in here.